### ISSUE TITLE
Adopt IProgressService for search / replace

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#74290]
**tl;dr;**: allow to pass in `IProgress<IProgressStep>` to [`BulkEdit`](https://github.com/Microsoft/vscode/blob/029d92b68dc51b42a65cf58bf8f501cab7d90f28/src/vs/workbench/services/bulkEdit/browser/bulkEditService.ts#L230)

Today we have `IProgressService2` (which I just renamed to `IProgressService`) and `IProgressService` (which I just renamed to `ILocalProgressService`). The former is the "proper" one that allows to report progress into a view location and should be used in almost all cases. I would like the latter (`ILocalProgressService`) to only be used within an editor (and then possibly rename the service to something else, e.g. `IEditorProgressService`). We still need that one for the standalone case too.

I would like us to adopt `IProgressService2` in more places and already pushed a change to adopt it for the explorer (https://github.com/microsoft/vscode/commit/1ce1371dc3393a4f3bc50852c0840f4ea3c54717). There is one remaining use in search that needs a bit of tweaking. Specifically the search/replace feature seems to pass on a `IProgressRunner` to the bulk edit service eventually to report accurate progress (`total` / `worked`). It would be great if that could change to pass in a `IProgress<IProgressStep>` which then would allow to use it with the proper `IProgressService`. However, we seem to set the total amount of work from within the bulk edit:

https://github.com/microsoft/vscode/blob/4ea0603c7c6d7d8ed93760913cd286da0cbd339c/src/vs/workbench/services/bulkEdit/browser/bulkEditService.ts#L298

We either need a way to know the total work upfront or allow for the `IProgress<IProgressStep>` to change the `total` after the fact.

The other usage of the local progress service is easier to adapt: https://github.com/microsoft/vscode/commit/029d92b68dc51b42a65cf58bf8f501cab7d90f28

**PS:** one visible difference of this change is that the we will draw a progress badge over the viewlet. This imho is the right thing to do because it makes progress reporting in views consistent with other views.

![before](https://user-images.githubusercontent.com/900690/58342947-27b37080-7e52-11e9-8d25-da22b98b65ac.gif)

**PPS:** it would be great if even for the normal search we could report accurate progress. Not sure if RipGrep provides this at all somehow.

**Comment by roblourens** [comment]
> PPS: it would be great if even for the normal search we could report accurate progress. Not sure if RipGrep provides this at all somehow.

Nope
